packages/c-pin-input/src/use-pin-input.ts

```diff
@@ -0,0 +1,43 @@
+import { computed } from "vue"
+import { connect, machine } from "@zag-js/pin-input"
+import { normalizeProps, useMachine } from "@zag-js/vue"
+
+export interface PinInputProps {
+  id: string
+  value: string[]
+  placeholder: string
+  type: "alphanumeric" | "numeric"
+  otp: boolean
+  mask: boolean
+  blurOnComplete: boolean
+}
+
+export const usePinInputMachine = (
+  params: PinInputProps,
+  emit: CallableFunction
+) => {
+  const [state, send] = useMachine(
+    machine({
+      id: params.id,
+      value: params.value,
+      placeholder: params.placeholder,
+      type: params.type,
+      otp: params.otp,
+      mask: params.mask,
+      blurOnComplete: params.blurOnComplete,
+      onChange: (value) => {
+        emit("change", value)
```

Comment:
I think we can add support two way binding for the pin-input component using `v-model`. Perhaps at this `onChange` callback we can also call
```suggestion
        emit("change", value)
        emit("update:modelValue", value)
```